### PR TITLE
GraphQL schema + adapter updates to handle GoQuorum transaction fields

### PR DESCRIPTION
**Pull request by mark-terry**
## PR description
Updates the GraphQL schema + adapter to handle GoQuorum transaction fields

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java [hunk 1]
```diff
@@ -259,7 +275,46 @@ public GraphQLDataFetchers(final Set<Capability> supportedCapabilities) {
...
+    final GoQuorumEnclave enclave = goQuorumPrivacyParameters.get().enclave();
+    final Bytes enclavePayload;
+
+    try {
+      // Retrieve the payload from the enclave
+      enclavePayload =
+          Bytes.wrap(enclave.receive(currTransaction.getPayload().toBase64String()).getPayload());
+
+      // Create a new transaction containing the retrieved payload
+      final Transaction newTransaction =
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java:300]: I don't like the names currTransaction and newTransaction. Can we make currTransaction -> transaction and make the new Transaction call inline so it doesn't need a name

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java [hunk 2]
```diff
@@ -259,7 +275,46 @@ public GraphQLDataFetchers(final Set<Capability> supportedCapabilities) {
...
+              currTransaction.getGasLimit(),
+              currTransaction.getTo(),
+              currTransaction.getValue(),
+              currTransaction.getSignature(),
+              enclavePayload,
+              currTransaction.getSender(),
+              currTransaction.getChainId(),
+              Optional.ofNullable(currTransaction.getV()));
+      return new TransactionAdapter(new TransactionWithMetadata(newTransaction));
+    } catch (final Exception ex) {
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java:315]: can we be more specific about what this exception is?

**Review by macfarla — COMMENTED** [review]
Does this make the GQ AT pass?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java (hunk 1 above)
Comment (mark-terry) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java:300]: Happy to do that if you find it clearer.

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java (hunk 2 above)
Comment (mark-terry) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java:315]: The most common exception I'd expect here is an EnclaveClientException, but I'm not sure it's the only one.

**Review by macfarla — APPROVED** [review]

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/internal/pojoadapter/TransactionAdapter.java [hunk 3]
```diff
@@ -185,4 +185,16 @@ public TransactionAdapter(final TransactionWithMetadata transactionWithMetadata)
     }
     return results;
   }
+
+  public Optional<Boolean> getIsPrivate() {
```
Comment (pinges) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/internal/pojoadapter/TransactionAdapter.java:189]: does that really have to be an Optional? I think it should just return the boolean ...

**Pull request merged**